wx/lib/asyncio.py

```diff
@@ -0,0 +1,336 @@
+finishes. Additionally, as all coroutines (all coroutines launched in the main event
+loop, exactly) will be executed in the main thread, there will be no worry about any
+confilction caused by concurrency in the program logic.
+
+Usage
+=====
+Sample usage::
+    #!/usr/bin/env python3
+    import asyncio
+    import threading
+    import time
+
+    import wx
+    import wx.lib.asyncio
+
+    def main():
+        asyncio.get_event_loop().set_debug(True)
+
+        def _another_loop_thread():
+            nonlocal another_loop
+            another_loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(another_loop)
+            another_loop.run_forever()
+            another_loop.close()
+
+        another_loop = None
+        another_loop_thread = threading.Thread(target=_another_loop_thread)
+        another_loop_thread.start()
+
+        def on_close(event):
+            another_loop.call_soon_threadsafe(another_loop.stop)
+            frame.Destroy()
+
+        frame = wx.Frame(None, title='Coroutine Integration in wxPython', size=wx.Size(800, 600))
+        frame.Bind(wx.EVT_CLOSE, on_close)
+        frame.CreateStatusBar()
+        frame.GetStatusBar().StatusText = 'Ready'
+        counter = 1
+
+        async def on_click(event):
+            def log(message: str):
+                frame.GetStatusBar().StatusText = message
+                print(message)
+
+            nonlocal counter
+            count = ' [' + str(counter) + ']'
+            counter += 1
+            log('Starting the event handler' + count)
+            await asyncio.sleep(1)  # Sleep in the current event loop
+            log('Running in the thread pool' + count)
+            # time.sleep is used to emulate synchronous time-consuming tasks
+            await asyncio.get_event_loop().run_in_executor(None, time.sleep, 1)
+            log('Running in another loop' + count)
+            # Socket operations are theoretically unsupported in WxEventLoop
+            # So a default event loop in a separate thread is sometime required
+            # asyncio.sleep is used to emulate these asynchronous tasks
+            await asyncio.wrap_future(asyncio.run_coroutine_threadsafe(asyncio.sleep(1), another_loop))
+            log('Ready' + count)
+
+        button = wx.Button(frame, label='\n'.join([
+            'Click to start the asynchronous event handler',
+            'The application will remain responsive while the handler is running',
+            'Try click here multiple times to launch multiple coroutines',
+            'These coroutines will not conflict each other as they are all in the same thread',
+        ]))
+        button.BindAsync(wx.EVT_BUTTON, on_click)
+
+        frame.Show()
+
+        asyncio.get_event_loop().run_forever()
+        another_loop_thread.join()
+
+
+    if __name__ == '__main__':
+        asyncio.set_event_loop_policy(wx.lib.asyncio.WxEventLoopPolicy(app=wx.App))
+        main()
+
+"""
+
+#!/usr/bin/env python3
```

Comment:
Remove this line. It only has effect as the first line in the file, and probably should not be in library modules anyway.

Reply:
Yes, the shebang should only reference itself on the firstline. Remove this otherwise you could have a possible security division/injection issue.
```if True: run()``` proves this as theta can be compiled into a "interpreter"